custom_components/abb_powerone_pvi_sunspec/const.py

```diff
@@ -61,9 +61,6 @@
     "AC_Power": ["AC Power", "acpower", UnitOfPower.WATT, "mdi:solar-power", SensorDeviceClass.POWER, STATE_CLASS_MEASUREMENT],
     "AC_Frequency": ["AC Frequency", "acfreq", UnitOfFrequency.HERTZ, "mdi:sine-wave", None, STATE_CLASS_MEASUREMENT],
     "Total_Energy": ["Total Energy", "totalenergy", UnitOfEnergy.WATT_HOUR, "mdi:solar-power", SensorDeviceClass.ENERGY, STATE_CLASS_TOTAL_INCREASING],
-    "DC_Curr": ["DC Current", "dccurr", UnitOfElectricCurrent.AMPERE, "mdi:current-ac", SensorDeviceClass.CURRENT, STATE_CLASS_MEASUREMENT],
```

Comment:
What do you think if we adopt your same approach for single-phase vs three-phase?

As you can see, single phase sensors is a subset of three-phase. We could have a SENSOR_TYPES_COMMON and if inv_type is 3 then add SINGLE_PHASE_THREE_PHASE sensors.

What do you think?

Reply:
yes, I have just checked for the items that are not available on my inverter, I will check and create this COMMON definition, and push again.

Reply:
These ones should be only in three-phase models: triple-check before committing. :)

```
    "AC_CurrentA": ["AC Current A", "accurrenta", UnitOfElectricCurrent.AMPERE, "mdi:current-ac", SensorDeviceClass.CURRENT, STATE_CLASS_MEASUREMENT],
    "AC_CurrentB": ["AC Current B", "accurrentb", UnitOfElectricCurrent.AMPERE, "mdi:current-ac", SensorDeviceClass.CURRENT, STATE_CLASS_MEASUREMENT],
    "AC_CurrentC": ["AC Current C", "accurrentc", UnitOfElectricCurrent.AMPERE, "mdi:current-ac", SensorDeviceClass.CURRENT, STATE_CLASS_MEASUREMENT],
    "AC_VoltageAB": ["AC Voltage AB", "acvoltageab", UnitOfElectricPotential.VOLT, "mdi:lightning-bolt", SensorDeviceClass.VOLTAGE, STATE_CLASS_MEASUREMENT],
    "AC_VoltageBC": ["AC Voltage BC", "acvoltagebc", UnitOfElectricPotential.VOLT, "mdi:lightning-bolt", SensorDeviceClass.VOLTAGE, STATE_CLASS_MEASUREMENT],
    "AC_VoltageCA": ["AC Voltage CA", "acvoltageca", UnitOfElectricPotential.VOLT, "mdi:lightning-bolt", SensorDeviceClass.VOLTAGE, STATE_CLASS_MEASUREMENT],
    "AC_VoltageBN": ["AC Voltage BN", "acvoltagebn", UnitOfElectricPotential.VOLT, "mdi:lightning-bolt", SensorDeviceClass.VOLTAGE, STATE_CLASS_MEASUREMENT],
    "AC_VoltageCN": ["AC Voltage CN", "acvoltagecn", UnitOfElectricPotential.VOLT, "mdi:lightning-bolt", SensorDeviceClass.VOLTAGE, STATE_CLASS_MEASUREMENT],

```

Reply:
The current A like voltage AN can be common, but this will break for current users, in this case I think is better to keep separate currentA from current in single phase

Reply:
`AC_Current` is in both
`AC_VoltageAN` is in both
`AC_CurrentA` is only in 3-phase

Why should it break for current users if we move `AC_CurrentA` in `SINGLE_PHASE_THREE_PHASE`and the others in `SENSOR_TYPES_COMMON`?

Reply:
In the 3 phase the AC_Current is combination os A B and C ?

as the voltage is defined as AN, on single phase the phase can be A also. But changing name will break for current users.

Reply:
AC_Current: Yes, it should be an aggregate. I'll check tomorrow when inverter is on.

> as the voltage is defined as AN, on single phase the phase can be A also. But changing name will break for current users.

AC_VoltageAN is present in both, so I don't understand what you have to rename. It's in common now, so it's fine both for monophase and 3-phase.

Basically the common section is what monophase represents, and if the inverter is 3-phase we add the sensors that are only present for 3-phase inverters. Am I missing something?